lib/index.js

```diff
@@ -100,11 +100,12 @@ exports.fetch = function (repo, opts) {
 }
 
 exports.extract = function (file, dest, opts) {
-	file = exists(file);
+  opts = opts || {};
+	file = exists(file, opts.name);
 	dest = path.resolve(dest || '.');
 	return new Promise((res, rej) => {
 		const ok = _ => res(dest);
-		opts = Object.assign({ strip:1 }, opts, { file, cwd:dest });
+		opts = Object.assign({ strip:1 }, opts, { file, cwd:dest, name: undefined });
```

Comment:
This would _have_ to be 

```js
opts = Object.assign({ strip:1, name:null }, opts, { file, cwd:dest });
```

Otherwise you're _always_ overwriting a user-specified `opts.name` with `name:undefined`.

> Used `null` here just for shorthand; that's not the problem in this case. :D